spi/imx6ull-qspi/qspi.h

```diff
@@ -0,0 +1,81 @@
+/*
+ * Phoenix-RTOS
+ *
+ * i.MX 6ULL QSPI lib API
+ *
+ * Copyright 2023 Phoenix Systems
+ * Author: Hubert Badocha
+ *
+ * %LICENSE%
+ */
+
+#ifndef IMX6ULL_QSPI_API_H
+#define IMX6ULL_QSPI_API_H
+
+#include <stddef.h>
+#include <stdint.h>
+
+#define MAX_READ_LEN  0xffff
+#define MAX_WRITE_LEN 0xffff
+
+#define LUT_INSTR(code, pad, operand) ((((code)&0x3f) << 10) | (((pad)&0x03) << 8) | ((operand)&0xff))
+
+typedef enum {
+	lut_stop = 0,
+	lut_cmd,
+	lut_addr,
+	lut_dummy,
+	lut_mode,
+	lut_mode2,
+	lut_mode4,
+	lut_read,
+	lut_write,
+	lut_jmp_on_cs,
+	lut_addr_ddr,
+	lut_mode_ddr,
+	lut_mode2_ddr,
+	lut_mode4_ddr,
+	lut_read_ddr,
+	lut_write_ddr,
+	lut_data_learn
+} lut_code_t;
```

Comment:
Our guideline - we do not type each enum in new line. There's a problem with clang-format, you can use /* clang-format off/on */ to bypass that for now